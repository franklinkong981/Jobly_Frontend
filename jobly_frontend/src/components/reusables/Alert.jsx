import React from "react";
import {v4 as uuidv4} from "uuid";

function Alert({alertType = "danger", alertTexts = []}) {
  return (
    <div className={`alert alert-${alertType}`} role="alert">
      {alertTexts.map(error => (
        <p className="mb-0 small" key={uuidv4()}>
          {error}
        </p>
      ))}
    </div>
  );
}

export default Alert;