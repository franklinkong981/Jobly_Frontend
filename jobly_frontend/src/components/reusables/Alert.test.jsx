import React from "react";
import {render} from "@testing-library/react";

import Alert from "./Alert.jsx";
import { expect, it } from "vitest";

const successMessages = ["Everything is fine", "It all works!"];
const dangerMessages = ["There is an error", "stop and fix"];

it("success alert renders without crashing", function() {
  render(<Alert alertType="success" alertTexts={successMessages} />);
});

it("danger alert renders without crashing", function() {
  render(<Alert alertType="danger" alertTexts={dangerMessages} />);
});

it("matches snapshot for success", function() {
  const {asFragment} = render(<Alert alertType="success" alertTexts={successMessages} />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for danger", function() {
  const {asFragment} = render(<Alert alertType="danger" alertTexts={dangerMessages} />);
  expect(asFragment()).toMatchSnapshot();
});