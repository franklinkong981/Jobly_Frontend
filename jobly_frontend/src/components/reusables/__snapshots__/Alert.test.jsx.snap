// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="alert alert-danger"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      There is an error
    </p>
    <p
      class="mb-0 small"
    >
      stop and fix
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
    role="alert"
  >
    <p
      class="mb-0 small"
    >
      Everything is fine
    </p>
    <p
      class="mb-0 small"
    >
      It all works!
    </p>
  </div>
</DocumentFragment>
`;
